lib/manager/cocoapods/readme.md

```diff
@@ -0,0 +1,232 @@
+
+Package manager used for applications only
+
+---
+
+#### If ranges are supported, are there any cases when Renovate should pin ranges to exact versions if rangeStrategy=auto?
+
+I don't see such reason.
+
+## Lookup
+
+#### Is a new datasource required? Provide details
+
+There is a index service for cocoapods. Developers always search for pod on this [site](https://github.com/CocoaPods):
+
+Also, all podspec are stored in one [repo](https://github.com/CocoaPods/Specs)
+
+---
+
+#### Will users need the capability to specify a custom host/registry to look up? Can it be found within the package files, or within other files inside the repository, or would it require Renovate configuration?
+
+User can provide remove git url ([details](https://guides.cocoapods.org/using/the-podfile.html#from-a-podspec-in-the-root-of-a-library-repo))
+
+`pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git'`
+
+---
+
+#### Do the package files contain any "constraints" on the parent language (e.g. supports only v3.x of Python) or platform (Linux, Windows, etc) that should be used in the lookup procedure?
+
+Cocoapods supports only mac platform.
+
+---
+
+#### Will users need the ability to configure language or other constraints using Renovate config?
+
+I would add an option to check dependencies with only specific swift version.
+
+## Artifacts
+
+#### Are lock files or checksum files used? Mandatory?
+
+Yes, lock file will be named `Podfile.lock`.
+
+---
+
+#### If so, what tool and exact commands should be used if updating 1 or more package versions in a dependency file?
+
+To update single package: `pod update SomePodName`
+
+To update all packages: `pod update`
+
+---
+
+#### If applicable, describe how the tool maintains a cache and if it can be controlled via CLI or env? Do you recommend the cache be kept or disabled/ignored?
+
+Cocoapods keeps cache in directory `${HOME}/Library/Caches/CocoaPods`
+
+You can get information about cache state and clean cache throutgh cocoapods ([details](https://guides.cocoapods.org/terminal/commands.html#group_cache))
+
+---
+
+#### If applicable, what command should be used to generate a lock file from scratch if you already have a package file? This will be used for "lock file maintenance".
+
+You can generate `Podfile.lock` by running `pod install`. This command also install required dependencies and link them to xcode project.
+
+I don't know hot to generate only `Podfile.lock`
+
+## Other
+
+#### Is there anything else to know about this package manager?
+
+Cocoapods has interesting feature that may be usefull for renovate.
+Command `pod outdated` lists all outdated dependecies ([details](https://guides.cocoapods.org/using/pod-install-vs-update.html#pod-outdated))
+
+---
+
+In order to add dependencies to your xcode project you have to create file named `Podfile`
+in your Xcode project directory
+
+```
+platform :ios, '8.0'
```

Comment:
This 'ios' and '8.0' is what I was referring to earlier in questions about platforms. I assume that packages themselves specify which ios versions they support so we need to match them?

Reply:
yes, packages specify which platform version they support

Reply:
I assume they can support multiple iOS versions, so we need to document how that syntax works.

Reply:
Here is reference with syntax info: https://guides.cocoapods.org/syntax/podspec.html#deployment_target